### PR TITLE
Fix exception in ReagentPrototype caused by IMetabolizable

### PR DESCRIPTION
**Pull request by Moneyl**
Just noticed that #437 causes exceptions in `Robust.Client` when `ReagentPrototype` tries to read the `IMetabolizable`s for each reagent. The exception happens when the client tries to find concrete implementations of `IMetabolizable` that are in `Content.Server`. It fixes this by checking via reflection what assembly is loading the prototype, and only trying to load the metabolizables if in `Robust.Server`.

The fix seems like a flimsy way to handle this, but it's the best I could come up with. All ears for a better way to handle this.

I don't want to move `ReagentPrototype` completely into `Content.Server` since it's useful for the client to view descriptions and values of reagents without needing to send that data over the network. The reagent dispenser GUI already does this. Ideally we also won't split the metabolizables list into it's own prototype that's only loaded by the server to keep things from getting too fragmented.

### CODE REVIEW

File: Content.Shared/Chemistry/ReagentPrototype.cs [hunk 1]
```diff
@@ -33,7 +33,15 @@ public void LoadFrom(YamlMappingNode mapping)
             serializer.DataField(ref _name, "name", string.Empty);
             serializer.DataField(ref _description, "desc", string.Empty);
             serializer.DataField(ref _substanceColor, "color", Color.White);
-            serializer.DataField(ref _metabolism, "metabolism", new List<IMetabolizable>{new DefaultMetabolizable()});
+
+            if (System.Reflection.Assembly.GetEntryAssembly()?.GetName().Name == "Robust.Server")
```
Comment (PJB3005) [inline Content.Shared/Chemistry/ReagentPrototype.cs]: This really isn't a good way to check this. It won't, for example, work in integration tests.

I'd rather you make some kind of IoC-based thing to store this.

Comment (Moneyl) [reply Content.Shared/Chemistry/ReagentPrototype.cs]: Done.

File: Content.Shared/Interfaces/IModuleManager.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    /// Provides a simple way to check whether calling code is being run by
+    /// Robust.Client, or Robust.Server. Useful for code in Content.Shared
+    /// that wants different behavior depending on if client or server is using it.
+    /// </summary>
+    public interface IModuleManager
+    {
+        /// <summary>
+        /// Returns true if the code is being run by the client, returns false otherwise.
+        /// </summary>
+        bool IsClientModule();
```
Comment (PJB3005) [inline Content.Shared/Interfaces/IModuleManager.cs]: Make this a property?

Comment (Moneyl) [reply Content.Shared/Interfaces/IModuleManager.cs]: Done. Didn't know you could have properties in interfaces. Thought it was functions only.

Comment (PJB3005) [reply Content.Shared/Interfaces/IModuleManager.cs]: Properties are for all intents and purposes just functions with some syntax sugar, so yeah.

**Pull request merged**